site/tests/app/libraries/routers/WebRouterTester.php

```diff
@@ -39,9 +39,18 @@ public function testRedirectToLoginFromCourse() {
         $request = Request::create(
             "/s19/sample"
         );
+        $old_request_url = $request->getUriForPath($request->getPathInfo());
+
+        $query_obj = $request->query->all();
+        $filtered_query_obj = array_filter($query_obj, function ($k) {
+            return $k !== "url";
+        }, ARRAY_FILTER_USE_KEY);
+
+        $query_string = empty($filtered_query_obj) ? null : '?' . http_build_query($filtered_query_obj);
+
         $response = WebRouter::getWebResponse($request, $core);
         $this->assertEquals(
-            $core->buildUrl(['authentication', 'login']) . '?old=' . urlencode($request->getUriForPath($request->getPathInfo())),
+            $core->buildUrl(['authentication', 'login']) . '?old=' . urlencode($old_request_url . $query_string),
```

Comment:
Welp, this was missed in my initial review of this. For both tests, can we replace the usage of a variable created off the `$request` variable to just a set hard-coded string. This will make it easier to reason about what it's doing and catch if behavior does subtle change going forward.

Reply:
Yes, I will keep this in mind for future PRs